draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -654,476 +526,437 @@
+	However, as with deletion of a "pending" trigger, dCDN does not guarantee this.
+   </t>
    <t>
-   If the CI/T Command cannot be stopped immediately, the status in the
-   corresponding Trigger Status Resource MUST be set to "cancelling",
-   and the Trigger Status Resource MUST remain in the collection of
-   Trigger Status Resources for active CI/T Commands. If processing is
-   stopped before normal completion, the status value in the Trigger
-   Status Resource MUST be set to "cancelled", and the Trigger Status
-   Resource MUST be included in the collection of failed CI/T Trigger
-   Commands.</t>
-
+	Deletion of a "complete", "cancelled" or "failed" trigger no processing in dCDN.
+   </t>
    <t>
-   Cancellation of a "complete" or "failed" Trigger Status Resource
-   requires no processing in dCDN. Its status MUST NOT be changed
-   to "cancelled".</t>
+	dCDN MUST respond to the trigger deletion request appropriately.
+   	dCDN MUST respond with the HTTP status code 200 ("OK") without a response body if the trigger has been deleted immediately. 
+   	202 ("Accepted") if the command has been accepted but the trigger has not been deleted yet. 
+   	404 ("Not Found") when the trigger resource does not exist, or 501 ("Not Implemented") if deletion is not supported by dCDN.
+   </t>
+   <t>
+	The trigger state MUST be set to "cancelling" while dCDN is processing a deletion request asynchronously.
+   </t>
 
+   	
    </section>
 
-   <section title="Deleting Triggers" anchor="deleting-triggers"><t>
-   uCDN can delete Trigger Status Resources at any time using the
-   HTTP DELETE method. As defined in the CDNI Control Interface / Triggers 1st edition <xref target="RFC8007"/>,
-   the effect is similar to cancellation, but no
-   Trigger Status Resource remains afterwards. Note that in cases where the 
-   trigger's status before the deletion is not a terminal status (i.e., "completed", "failed"
-   or "cancelled"), uCDN would not have the ability to further monitor the 
-   cancellation process and the final status of the trigger. 
-   For this reason, it is recommended that the trigger is cancelled (see <xref target="cancelling-triggers"/>) 
-   before trigger deletion and deletion is only used on triggers with a terminal status.</t>
-
+   <section title="Expiry of Triggers" anchor="expiry-of-triggers">
    <t>
-   Once deleted, the references to a Trigger Status Resource MUST be
-   removed from all Trigger Status Resource collections. Subsequent
-   requests to GET the deleted Trigger Status Resource SHOULD be
-   rejected by dCDN with an HTTP error.</t>
-
+  	dCDN MAY automatically delete trigger resources sometime after they reach a terminate state (one of "complete", "processed", 
+   	"failed", or "cancelled"). In this case, after dCDN removed such trigger, it MUST respond to subsequent requests for it with 
+   	the HTTP error 404 ("Not Found") and remove it from the trigger collection resource.
```

Comment:
Instead of "the trigger collection resource", would it be more accurate to says, "all trigger collection resources"?